test/gtest/core/test_mc_reduce_cuda.cc

```diff
@@ -35,3 +35,24 @@ TYPED_TEST(test_mc_reduce, ucc_reduce_multi_cuda) {
         TypeParam::assert_equal(res, this->res_h[i]);
     }
 }
+
+TYPED_TEST(test_mc_reduce_alpha, ucc_reduce_multi_alpha_cuda) {
+    const int num_vec = 3;
+    const double alpha = 0.7;
+    this->alloc_bufs(UCC_MEMORY_TYPE_CUDA, num_vec);
+    ucc_mc_reduce_multi_alpha(this->buf1_d, this->buf2_d, this->res_d, num_vec,
+                              this->COUNT, this->COUNT*sizeof(*this->buf2_d),
+                              TypeParam::dt, TypeParam::redop, UCC_OP_PROD, alpha,
+							  UCC_MEMORY_TYPE_CUDA);
+    ucc_mc_memcpy(this->res_h, this->res_d, this->COUNT * sizeof(*this->res_d),
+                  UCC_MEMORY_TYPE_HOST, UCC_MEMORY_TYPE_CUDA);
+    for (int i = 0; i < this->COUNT; i++) {
+        typename TypeParam::type res = TypeParam::do_op(this->buf1_h[i],
+                                                        this->buf2_h[i]);
+        for (int j = 1; j < num_vec; j++) {
+            res = TypeParam::do_op(this->buf2_h[i + j * this->COUNT], res);
+        }
+        res *= (TypeParam::type)alpha;
```

Comment:
it will not work, compiler doesn't know whether TypeParam::type is type or something different